src/main/java/com/example/demo/role_service/RoleService.java

```diff
@@ -1,9 +1,41 @@
 package com.example.demo.role_service;
 
 import com.example.demo.entity.Role;
+import com.example.demo.exceptions.RoleNotFoundException;
+import com.example.demo.repository.RoleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
-public interface RoleService {
```

Comment:
to byl interfejs i byla klasa RoleServiceImpl. Unikamy takich nadmiarowych abstrakcji.
Nigdy nie zrobisz RoleServiceAnotherImpl. Ewentualnie na potrzeby testow, ale akurat klasy serwisow nie mockujemy tylko wlasnie testujemy. Wiec przy klasach serwisow nie trzeba wprowadzac interfejsow, to tylko dodaje zbednego kodu i utrudnia czytanie